third-party/realdds/include/realdds/dds-adapter-watcher.h

```diff
@@ -0,0 +1,40 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2024 Intel Corporation. All Rights Reserved.
+#pragma once
+
+#include <rsutils/os/adapter-watcher.h>
+#include <set>
+#include <string>
+
+
+namespace realdds {
+
+
+namespace detail {
+class adapter_watcher_singleton;
+}
+
+
+// Watch for changes to network adapter IPs
+// 
+// Unlike rsutils::os::adapter_watcher, we only call the callbacks when actual changes to IPs are made, which can
+// sometimes be seconds after adapter-change notifications are sent
+// 
+// All you have to do is create a watcher and keep a pointer to it to get notifications
+//
+class dds_adapter_watcher
+{
+    std::shared_ptr< detail::adapter_watcher_singleton > _singleton;
+    rsutils::subscription _subscription;
+
+public:
+    using callback = std::function< void() >;
+
+    dds_adapter_watcher( callback && );
```

Comment:
It's not a singleton if the Ctor is public, right?

each `dds_adapter_watcher` instance will create a new shared ptr?

Reply:
No, they'll all share a singleton -- see the ctor implementation.

Reply:
Will dive in